udfs/remote_udfs/nlp/README.md

```diff
@@ -0,0 +1,129 @@
+* location - the location for the external connection in BigQuery
+* project - the project where your external connection was created
+* connection_name - the name (or friendly/display name) of the external connection
+
+```
+bq show --connection --project_id=$PROJECT --location=$LOCATION $CONNECTION_NAME
+```
+
+Within properties you'll find *serviceAccountId* which will have the service ID you'll need in a subsequent step.
+
+### Deploying your Cloud Function
+
+[More information about deploying your cloud function can be found here.](https://cloud.google.com/functions/docs/deploy)
+
+Snippet provided below for brevity.  
+It’s recommended that you keep the default authentication instead of allowing unauthenticated invocation of your Cloud Function or Cloud Run service.
+
+* cf_name - the name of the cloud function
+* project - the project the cloud function is deployed to 
+* runtime - this was defaulted to python39 but can be changed as required 
+```
+gcloud functions deploy $CF_NAME \
+--project=$PROJECT --runtime=python39 --entry-point=$ENTRY_POINT --source=call_nlp --trigger-http
+```
+
+### Granting the service account invoker permissions on the functions
+
+Grant the service account obtained above permissions to invoke the functions.
+* project - the project where your external connection was created
+* cf_name - the name of your cloud function 
+* service_account - the service account obtained above
+
+**_NOTE:_** For the below command you will need to have jq installed. 
+
+```
+SERVICE_ACCOUNT=$(bq show --connection --project_id=$PROJECT --location=$LOCATION $CONNECTION_NAME | jq '.cloudResource.serviceAccountId' | tr -d '"')
+gcloud --project=$PROJECT functions add-iam-policy-binding $CF_NAME --member=serviceAccount:$SERVICE_ACCOUNT --role=roles/cloudfunctions.invoker
+```
+
+### Obtaining the full Cloud Function endpoint
+
+[More information about the gcloud functions describe can be found here.](https://cloud.google.com/sdk/gcloud/reference/functions/describe)
+* cf_name - the name of the cloud function
+```
+gcloud functions describe $CF_NAME
+```
+
+You will need the full URL under the httpsTrigger section.
+It should look something like this:
+```
+https://<location>-<project-id>.cloudfunctions.net/<cf-name>
+```
+
+### Creating your BigQuery UDF on BigQuery
+
+You can choose to do this many ways.  
+* [creating your bq function](/remote_udfs/nlp/create_bq_function.sh) 
+
+The input parameters for the helper script above need to be in order:
+* project - the project you deployed to 
+* dataset - the BigQuery dataset you want to deploy to 
+* location - the location where your BigQuery dataset is 
+* connection_name - the name of the connection 
+* endpoint - the full endpoint you obtained above
+* cf_name - the name of the cloud function deployed
+
+```
+ENDPOINT=$(gcloud functions describe $CF_NAME --format="value(httpsTrigger.url)")
+sh create_bq_function.sh $PROJECT $DATASET $LOCATION $CONNECTION_NAME $ENDPOINT
```

Comment:
CONNECTION_NAME needs to be in the following format: 
`project_id.location.connection_name`

[Source](https://cloud.google.com/bigquery/docs/reference/standard-sql/remote-functions#create_a_remote_function_2)

Reply:
good catch, thanks